panaroo/integrate.py

```diff
@@ -0,0 +1,272 @@
+    
+    core.add_argument("--core_threshold",
+                      dest="core",
+                      help="Core-genome sample threshold (default=0.95)",
+                      type=float,
+                      default=0.95) 
+    
+    graph = parser.add_argument_group('Graph correction')
+    
+    graph.add_argument("--all_seq_in_graph",
+                       dest="all_seq_in_graph",
+                       help=("Retains all DNA sequence for each gene cluster in the graph " +
+                             "output. Off by default as it uses a large amount of space."),
+                       action='store_true',
+                       default=False)
+        
+    # Other options
+    parser.add_argument("-t",
+                        "--threads",
+                        dest="n_cpu",
+                        help="number of threads to use (default=1)",
+                        type=int,
+                        default=1) 
+    
+    parser.add_argument("--quiet",
+                        dest="quiet",
+                        help="suppress additional output",
+                        action='store_true',
+                        default=False)
+        
+    parser.add_argument("--version'",
+                        action="version",
+                        version='%(prog)s ' + __version__)
+
+    args = parser.parse_args()
+    return (args)
+
+
+def reformat_network(single_gml, output_dir, isolateName): #Generate network output needs to be reformatted for single gff inputs
+    
+    for adj in single_gml._adj:
+        for x in single_gml._adj[adj]:
+            y = single_gml._adj[adj][x]
+        
+            y.pop('members')
+            zero = {'members': 0}
+            y.update(zero)
+        
+            genomes = {'genomeIDs' : '0'}
+            y.update(genomes)
+        
+    for node in single_gml._node:
+        y = single_gml._node[node]
+        y.pop('members')
+        
+        zero = {'members': 0} #members are assigned intbitset[0]. needs to be 0
+        y.update(zero)
+        
+        rep = "[']"
+        for char in rep:
+            y['centroid'] = (str(y['centroid'])).replace(char, '')
+            y['dna'] = (str(y['dna'])).replace(char, '')
+            y['protein'] = (str(y['protein'])).replace(char, '')
+        
+        y['hasEnd'] = int(y['hasEnd'])
+        y['mergedDNA'] = int(y['mergedDNA'])
+        y['paralog'] = int(y['paralog'])
+        
+        y['longCentroidID'] = list(y['longCentroidID'])
+        y['seqIDs'] = list(y['seqIDs'])
+    
+    single_gml.graph.update({'isolateNames' : 'x'}) # isolateName assigned x as list(x) in merged graph
+    nx.write_gml(single_gml, output_dir + "final_graph.gml")
+    
+    return single_gml
+
+
+def main(): #Takes a single GFF input, generates a graph and merges with a pre-existing graph
+    args = get_options()
+
+    # make sure trailing forward slash is present
+    args.output_dir = os.path.join(args.output_dir, "")
```

Comment:
Check that this directory exists, and create it if not

Reply:
code has been added but returns an error if the directory does not exist for unknown reasons